exporter/opentelemetry-exporter-zipkin/src/opentelemetry/exporter/zipkin/__init__.py

```diff
@@ -60,384 +64,101 @@
+opentelemetry.exporter.zipkin.encoder.Encoding for supported options.
 
 API
 ---
 """
 
-import json
 import logging
-from typing import Optional, Sequence, Union
-from urllib.parse import urlparse
+from typing import Optional, Sequence
 
 import requests
 
 from opentelemetry.configuration import Configuration
-from opentelemetry.exporter.zipkin.gen import zipkin_pb2
+from opentelemetry.exporter.zipkin.encoder import Encoder, Encoding
+from opentelemetry.exporter.zipkin.encoder.v1.json import JsonV1Encoder
+from opentelemetry.exporter.zipkin.encoder.v2.json import JsonV2Encoder
+from opentelemetry.exporter.zipkin.encoder.v2.protobuf import ProtobufEncoder
+from opentelemetry.exporter.zipkin.node_endpoint import IpInput, NodeEndpoint
 from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
-from opentelemetry.trace import Span, SpanContext, SpanKind
-from opentelemetry.trace.status import StatusCode
-
-TRANSPORT_FORMAT_JSON = "json"
-TRANSPORT_FORMAT_PROTOBUF = "protobuf"
-
-DEFAULT_RETRY = False
-DEFAULT_URL = "http://localhost:9411/api/v2/spans"
-DEFAULT_MAX_TAG_VALUE_LENGTH = 128
-
-SPAN_KIND_MAP_JSON = {
-    SpanKind.INTERNAL: None,
-    SpanKind.SERVER: "SERVER",
-    SpanKind.CLIENT: "CLIENT",
-    SpanKind.PRODUCER: "PRODUCER",
-    SpanKind.CONSUMER: "CONSUMER",
-}
+from opentelemetry.trace import Span
 
-SPAN_KIND_MAP_PROTOBUF = {
-    SpanKind.INTERNAL: zipkin_pb2.Span.Kind.SPAN_KIND_UNSPECIFIED,
-    SpanKind.SERVER: zipkin_pb2.Span.Kind.SERVER,
-    SpanKind.CLIENT: zipkin_pb2.Span.Kind.CLIENT,
-    SpanKind.PRODUCER: zipkin_pb2.Span.Kind.PRODUCER,
-    SpanKind.CONSUMER: zipkin_pb2.Span.Kind.CONSUMER,
-}
-
-NAME_KEY = "otel.library.name"
-VERSION_KEY = "otel.library.version"
-
-SUCCESS_STATUS_CODES = (200, 202)
+DEFAULT_SERVICE_NAME = "unknown"
+DEFAULT_ENDPOINT = "http://localhost:9411/api/v2/spans"
+DEFAULT_ENCODING = Encoding.V2_JSON
+REQUESTS_SUCCESS_STATUS_CODES = (200, 202)
 
 logger = logging.getLogger(__name__)
 
 
 class ZipkinSpanExporter(SpanExporter):
-    """Zipkin span exporter for OpenTelemetry.
-
-    Args:
-        service_name: Service that logged an annotation in a trace.Classifier
-            when query for spans.
-        url: The Zipkin endpoint URL
-        ipv4: Primary IPv4 address associated with this connection.
-        ipv6: Primary IPv6 address associated with this connection.
-        retry: Set to True to configure the exporter to retry on failure.
-        transport_format: transport interchange format to use
-    """
-
     def __init__(
         self,
-        service_name: str,
-        url: str = None,
-        ipv4: Optional[str] = None,
-        ipv6: Optional[str] = None,
-        retry: Optional[str] = DEFAULT_RETRY,
-        max_tag_value_length: Optional[int] = DEFAULT_MAX_TAG_VALUE_LENGTH,
-        transport_format: Union[
-            TRANSPORT_FORMAT_JSON, TRANSPORT_FORMAT_PROTOBUF, None
-        ] = None,
+        service_name: Optional[str] = None,
```

Comment:
Similar to earlier point. Probably we shouldn't have `service_name` argument for exporters.

Reply:
Removed!